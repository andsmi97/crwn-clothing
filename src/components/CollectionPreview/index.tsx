import React from "react";
import "./styles.scss";
import { IShopData } from "../../types";
import CollectionItem from "../CollectionItem";

export default ({ title, items }: IShopData): JSX.Element => {
  return (
    <div className="collection-preview">
      <h1 className="title">{title}</h1>
      <div className="preview">
        {items
          .filter((item, index) => index < 4)
          .map(({ id, ...restProps }) => (
            <CollectionItem key={id} id={id} {...restProps} />
          ))}
      </div>
    </div>
  );
};
