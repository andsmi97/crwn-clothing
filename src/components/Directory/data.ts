import { IMenuItem } from "../../types";
export const data: IMenuItem[] = [
  {
    title: "hats",
    imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
    id: 1,
    linkUrl: "shop/hats",
    size: "small"
  },
  {
    title: "jackets",
    imageUrl: "https://i.ibb.co/px2tCc3/jackets.png",
    id: 2,
    linkUrl: "shop/jackets",
    size: "small"
  },
  {
    title: "sneakers",
    imageUrl: "https://i.ibb.co/0jqHpnp/sneakers.png",
    id: 3,
    linkUrl: "shop/sneakers",
    size: "small"
  },
  {
    title: "womens",
    imageUrl: "https://i.ibb.co/GCCdy8t/womens.png",
    size: "large",
    id: 4,
    linkUrl: "shop/womens"
  },
  {
    title: "mens",
    imageUrl: "https://i.ibb.co/R70vBrQ/men.png",
    size: "large",
    id: 5,
    linkUrl: "shop/mens"
  }
];
