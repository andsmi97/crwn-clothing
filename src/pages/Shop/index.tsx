import React from "react";
import CollectionsOverview from "../../components/CollectionsOverview";
import CollectionPage from "../Collection";

import { Route } from "react-router-dom";

const Shop = ({ match }: any): JSX.Element => {
  return (
    <div className="shop-page">
      <Route exact path={match.path} component={CollectionsOverview} />
      <Route path={`${match.path}/:collectionId`} component={CollectionPage} />
    </div>
  );
};

export default Shop;
