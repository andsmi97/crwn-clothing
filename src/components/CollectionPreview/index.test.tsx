import React from "react";
import { shallow, mount } from "enzyme";
// import { expect } from "chai";
import CollectionPreview from "../CollectionPreview";
import { data } from "../../pages/Shop/data";

describe("<CollectionPreview />", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <CollectionPreview
        id={data[0].id}
        title={data[0].title}
        items={data[0].items}
        routeName={data[0].routeName}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
