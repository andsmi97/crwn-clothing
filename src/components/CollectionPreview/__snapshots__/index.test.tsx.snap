// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionPreview /> matches snapshot 1`] = `ShallowWrapper {}`;
